### PR TITLE
Add no new development notice to perf tool

### PR DESCRIPTION
**Pull request by VijayanB**
### Description
Added notice to inform developers to consider adding new features to OSB
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by VijayanB** [comment]
BWC test failure is not related to this PR

File: benchmarks/perf-tool/README.md [hunk 1]
```diff
@@ -1,3 +1,5 @@
+# IMPORTANT NOTE: No new features will be added to this tool . This tool is currently in maintanence mode. All new features will be added to [vector search workload]( https://github.com/opensearch-project/opensearch-benchmark-workloads/tree/main/vectorsearch)
```
Comment (navneet1v) [inline benchmarks/perf-tool/README.md:1]: can we add same in the OSB readme too?

Comment (VijayanB) [reply benchmarks/perf-tool/README.md:1]: Ack

**Review by navneet1v — APPROVED** [review]

**Comment by navneet1v** [comment]
> BWC test failure is not related to this PR

The BWC issue will go away once we upgrade to 2.14 SNAPSHOTS

**Review by junqiu-lei — APPROVED** [review]

**Pull request merged**